### DISCUSSION TITLE
question: is this crate relevant for some diff questions outlined

### DISCUSSION THREAD

**Discussion by dzmitry-lahoda** [Byron/treediff-rs#16] [Q&A]
I would like to discriminate next diffs:

- sequences
  - reorder [a,c,b] became [c,b,a] (different from sorted)
  - add/remove/insert (seems support)
  - not set anymore [a,b], became [a,a,b]
  - sorted vs unsorted [b,a] became [a,b]

- maps (sequences of key values entries, sequences of 2-tuples):
  - similar to sequences when applied
  - sorted by key
  - duplicate of key
  - add/remove/insert (seems support)
  - key vs value updates

- considering primitives (like u128) depending on context be primitives (single value in place eq update) vs being vec of boolean numbers

these scenarios are based on realworld diff and equality considerations,

also relevant https://github.com/Byron/treediff-rs/issues/5

**Comment by Byron** [comment]
Given the core algorithm here is very simple, it's probably best for many to implement what they need themselves.
You seem to know the requirements, and putting down a few `#[test]`s might be the best way to find out what's happening here.
